server/src/main/java/org/opensearch/cluster/routing/allocation/ExistingShardsAllocator.java

```diff
@@ -108,14 +112,26 @@ void allocateUnassigned(
      *
      * Allocation service will currently run the default implementation of it implemented by {@link ShardsBatchGatewayAllocator}
      */
-    default void allocateAllUnassignedShards(RoutingAllocation allocation, boolean primary) {
+    default BatchRunnableExecutor allocateAllUnassignedShards(RoutingAllocation allocation, boolean primary) {
         RoutingNodes.UnassignedShards.UnassignedIterator iterator = allocation.routingNodes().unassigned().iterator();
+        List<TimeoutAwareRunnable> runnables = new ArrayList<>();
         while (iterator.hasNext()) {
             ShardRouting shardRouting = iterator.next();
             if (shardRouting.primary() == primary) {
-                allocateUnassigned(shardRouting, allocation, iterator);
+                runnables.add(new TimeoutAwareRunnable() {
+                    @Override
+                    public void onTimeout() {
+                        // do nothing
+                    }
+
+                    @Override
+                    public void run() {
+                        allocateUnassigned(shardRouting, allocation, iterator);
+                    }
+                });
             }
         }
+        return new BatchRunnableExecutor(runnables, () -> TimeValue.MINUS_ONE);
```

Comment:
this will not timeout anything, what is the purpose of using BatchRunnableExecutor here?

Reply:
This is default implementation, and BatchRunnableExecutor is necessary for `ShardsBatchGatewayAllocator`. `ShardsBatchGatewayAllocator` overrides this method to implement timeout handling and allocation logic

Reply:
Simplified this